clash-prelude/src/Clash/XException.hs

```diff
@@ -6,14 +6,58 @@ Copyright  :  (C) 2016,      University of Twente,
 License    :  BSD2 (see the file LICENSE)
 Maintainer :  QBayLogic B.V. <devops@qbaylogic.com>
 
-'XException': An exception for uninitialized values
+This module houses utilities to deal with /undefined/ values in Clash. An undefined
+value is translated to @x@ in VHDL and Verilog. At its core, this is handled by
+a special 'Exception' called 'XException'.
 
->>> show (errorX "undefined" :: Integer, 4 :: Int)
-"(*** Exception: X: undefined
-CallStack (from HasCallStack):
+Like any 'Exception' in Haskell, any term can take on this value:
+
+>>> error "help" :: Int
+*** Exception: help
+...
+>>> errorX "help" :: Int
+*** Exception: X: help
 ...
->>> showX (errorX "undefined" :: Integer, 4 :: Int)
-"(undefined,4)"
+
+Within the Haskell ecosystem, and throughout this module, you will find these
+exception values being referred to as /bottom/. This is sometimes written as a
+symbol: @_|_@.
+
+All 'Exception's, including 'XException', are \"infectious\". Any value depending
+on an 'Exception' will itself be replaced by it:
+
+>>> let x = error "help" :: Int
+>>> x + 1
+*** Exception: help
+
+In typical non-IO Haskell, exceptions are generally used to halt execution in case
+of /the impossible/. E.g., situations where GHC can't proof a branch won't be
+evaluated, but it nevertheless won't:
+
+@
+if | a < 0     = "less than"
+   | a == 0    = "equal"
+   | a > 0     = "greater than"
+   | otherwise = error "Can't happen"
+@
+
+In Clash they are used far more pervasively, as hardware architectures often
+deal with /undefinedness/. For example, the initial value of a 'dflipflop' is
```

Comment:
Let's not use undefinedness. It's not really a proper word, and it reads clunkily IMO. I can't think of a good single word to take it's place

Reply:
"Unknown values"? "Undefined values"?